src/base/private/profile_p.cpp

```diff
@@ -0,0 +1,208 @@
+        result = Utils::Fs::fromNativePath(QString::fromWCharArray(path));
+    if (!QCoreApplication::applicationName().isEmpty())
+        result += QLatin1String("/") + qApp->applicationName();
+#elif defined(Q_OS_MAC)
+    FSRef ref;
+    OSErr err = FSFindFolder(kUserDomain, kApplicationSupportFolderType, false, &ref);
+    if (err)
+        return QString();
+    QByteArray ba(2048, 0);
+    if (FSRefMakePath(&ref, reinterpret_cast<UInt8 *>(ba.data()), ba.size()) == noErr)
+        result = QString::fromUtf8(ba).normalized(QString::NormalizationForm_C);
+    result += QLatin1Char('/') + qApp->applicationName();
+#else
+    QString xdgDataHome = QLatin1String(qgetenv("XDG_DATA_HOME"));
+    if (xdgDataHome.isEmpty())
+        xdgDataHome = QDir::homePath() + QLatin1String("/.local/share");
+    xdgDataHome += QLatin1String("/data/")
+            + qApp->applicationName();
+    result = xdgDataHome;
+#endif
+    if (!result.endsWith("/"))
+        result += "/";
+    return result;
+}
+
+QString Private::DefaultProfile::downloadLocation() const
+{
+#if defined(Q_OS_WIN)
+    if (QSysInfo::windowsVersion() <= QSysInfo::WV_XP)  // Windows XP
+        return QDir(QStandardPaths::writableLocation(QStandardPaths::DocumentsLocation)).absoluteFilePath(
+            QCoreApplication::translate("fsutils", "Downloads"));
+#endif
+    return QStandardPaths::writableLocation(QStandardPaths::DownloadLocation);
+}
+
+SettingsPtr Private::DefaultProfile::applicationSettings(const QString &name) const
+{
+#if defined(Q_OS_WIN) || defined(Q_OS_MAC)
+    return SettingsPtr(new QSettings(QSettings::IniFormat, QSettings::UserScope, configurationName(), name));
+#else
+    return SettingsPtr(new QSettings(configurationName(), name));
+#endif
+}
+
+Private::CustomProfile::CustomProfile(const QString &rootPath, const QString &configurationName)
+    : Profile {configurationName}
+    , m_rootDirectory {QDir(rootPath).absoluteFilePath(this->configurationName())}
+{
+}
+
+QString Private::CustomProfile::baseDirectory() const
+{
+    return m_rootDirectory.canonicalPath();
+}
+
+QString Private::CustomProfile::cacheLocation() const
+{
+    return m_rootDirectory.absoluteFilePath(QLatin1String(cacheDirName));
+}
+
+QString Private::CustomProfile::configLocation() const
+{
+    return m_rootDirectory.absoluteFilePath(QLatin1String(configDirName));
+}
+
+QString Private::CustomProfile::dataLocation() const
+{
+    return m_rootDirectory.absoluteFilePath(QLatin1String(dataDirName));
+}
+
+QString Private::CustomProfile::downloadLocation() const
+{
+    return m_rootDirectory.absoluteFilePath(QLatin1String(downloadsDirName));
+}
+
+SettingsPtr Private::CustomProfile::applicationSettings(const QString &name) const
+{
+    // here we force QSettings::IniFormat format always because we need it to be portable across platforms
+    const QString settingsFileName {QDir(configLocation()).absoluteFilePath(name + QLatin1String(".conf"))};
```

Comment:
Disagree with you: current files have '.conf' extension, and if we use '.ini' in the custom profiles, we make a custom profile dir incompatible with the current one, while otherwise profiles dirs can be copied one into another.